x-pack/plugins/enterprise_search/public/applications/__mocks__/expected_async_error.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const expectedAsyncError = async (promise: Promise<unknown>) => {
```

Comment:
I like this a lot. I was imagining it'd have to be something more complex, but this is laser focused.

Reply:
Awesome! We can definitely extend it in the future if we need to is my thought, but maybe we won't have to.